### PR TITLE
Adding the billing email verified state to useremail

### PR DESCRIPTION
**Pull request by corps**
Sentry side of changes ahead of new billing notifications.

We capture the notion that a user's email has been 'verified' to receive billing notifications.  Once a user's email has been verified, it applies to all the organizations that belong to that they are either admin or owner of.

### CODE REVIEW

File: src/sentry/migrations/0608_add_billing_email_verified.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    is_dangerous = False
+
+    dependencies = [
+        ("sentry", "0607_drop_externalactor_actorid"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="useremail",
+            name="is_billing_notification_verified",
```
Comment (vbro) [inline src/sentry/migrations/0608_add_billing_email_verified.py:28]: Is it possible to rename this, and/or make this feature general purpose (i.e. not only used for billing)?
We usually avoid exposing billing and usage concepts in `sentry`.

File: src/sentry/models/useremail.py [hunk 2]
```diff
@@ -140,3 +148,18 @@ def write_relocation_import(
         # `--merge_users=true` case is handled in the `normalize_before_relocation_import()` method
         # above).
         return (useremail.pk, ImportKind.Inserted)
+
+    @classmethod
+    def find_billing_emails_for(cls, organization_id: int) -> List[UserEmail]:
```
Comment (vbro) [inline src/sentry/models/useremail.py:153]: This method also deals with billing. Could this be moved to `getsentry`?

File: src/sentry/migrations/0608_add_billing_email_verified.py (hunk 1 above)
Comment (corps) [reply src/sentry/migrations/0608_add_billing_email_verified.py:28]: Hmm, that's a good point.  The other way to model this is with a separate model that only exists in `getsentry` that relates back to `UserEmail`.

File: src/sentry/models/useremail.py [hunk 3]
```diff
@@ -140,3 +148,18 @@ def write_relocation_import(
...
+        from sentry.models.organizationmembermapping import OrganizationMemberMapping
+
+        admin_and_owners = OrganizationMemberMapping.objects.filter(
+            role__in=["admin", "owner"], organization_id=organization_id
```
Comment (mbauer404) [inline src/sentry/models/useremail.py:157]: Should this be `role__in=["billing", "owner"]`

File: src/sentry/migrations/0608_add_billing_email_verified.py (hunk 1 above)
Comment (markstory) [reply src/sentry/migrations/0608_add_billing_email_verified.py:28]: A separate table could be useful as you could also store the time of the most recent 'verification' for the email receipt and would also address having the billing role present in sentry.

File: src/sentry/models/useremail.py (hunk 3 above)
Comment (corps) [reply src/sentry/models/useremail.py:157]: Good call out -- I was told "admin", "owner", so I just triple check this.

**Comment by corps** [comment]
Replaced by https://github.com/getsentry/getsentry/pull/12188

**Pull request closed without merging**